hub/api/dataset.py

```diff
@@ -140,6 +140,45 @@ def empty(
             storage=cache_chain, read_only=read_only, public=public, token=token
         )
 
+    @staticmethod
+    def exists(
+        path: str,
+        memory_cache_size: int = DEFAULT_MEMORY_CACHE_SIZE,
+        local_cache_size: int = DEFAULT_LOCAL_CACHE_SIZE,
+        creds: Optional[dict] = None,
+        token: Optional[str] = None,
+    ) -> bool:
+        """Check if a dataset already exists
+
+        Args:
+            path (str): The full path to the dataset. Can be:-
+                - a Hub cloud path of the form hub://username/datasetname. To write to Hub cloud datasets, ensure that you are logged in to Hub (use 'activeloop login' from command line)
+                - an s3 path of the form s3://bucketname/path/to/dataset. Credentials are required in either the environment or passed to the creds argument.
+                - a local file system path of the form ./path/to/dataset or ~/path/to/dataset or path/to/dataset.
+                - a memory path of the form mem://path/to/dataset which doesn't save the dataset but keeps it in memory instead. Should be used only for testing as it does not persist.
+            memory_cache_size (int): The size of the memory cache to be used in MB.
+            local_cache_size (int): The size of the local filesystem cache to be used in MB.
+            creds (dict, optional): A dictionary containing credentials used to access the dataset at the path.
+                This takes precedence over credentials present in the environment. Currently only works with s3 paths.
+                It supports 'aws_access_key_id', 'aws_secret_access_key', 'aws_session_token', 'endpoint_url' and 'region' as keys.
+            token (str, optional): Activeloop token, used for fetching credentials for Hub datasets. This is optional, tokens are normally autogenerated.
+
+        Returns:
+            Boolean to indicate if dataset exists.
+        """
+        if creds is None:
+            creds = {}
+
+        storage, cache_chain = get_storage_and_cache_chain(
+            path=path,
+            read_only=False,
```

Comment:
read_only should probably be set to True, we're just checking existence so we don't need write access.